runtime/service.go

```diff
@@ -647,12 +648,16 @@ func (s *service) buildVMConfiguration(req *proto.CreateVMRequest) (*firecracker
 
 	var driveBuilder firecracker.DrivesBuilder
 	// Create non-stub drives
+	var path = s.config.RootDrive
 	if root := req.RootDrive; root != nil {
-		driveBuilder = firecracker.NewDrivesBuilder(root.PathOnHost)
-	} else {
-		driveBuilder = firecracker.NewDrivesBuilder(s.config.RootDrive)
+		path = root.PathOnHost
 	}
 
+	driveBuilder = firecracker.DrivesBuilder{}.
+		WithRootDrive(path, func(d *models.Drive) {
+			d.IsReadOnly = firecracker.Bool(true)
+		})
```

Comment:
Users are able to specify that their [`RootDrive`](https://github.com/firecracker-microvm/firecracker-containerd/blob/master/proto/firecracker.proto#L22) should be [read-only or read-write](https://github.com/firecracker-microvm/firecracker-containerd/blob/master/proto/types.proto#L111), so don't we want to only set the default of `IsReadOnly = true` when the user didn't actually provide a root drive config? Unless I'm misunderstanding this will always set the root drive to be readonly regardless of the configuration provided in the CreateVM request.

Reply:
Right. We are discarding the most of the RootDrive field on a request, but it shouldn't be.

Reply:
The last revision honors the fields from the request.